runtime-sdk/src/module.rs

```diff
@@ -276,8 +276,8 @@ pub trait AuthHandler {
         _ctx: &mut C,
         _tx: &Transaction,
     ) -> Result<(), modules::core::Error> {
-        // Default implementation doesn't do any checks.
-        Ok(())
+        // Default implementation rejects all transactions.
+        Err(modules::core::Error::NotAuthenticated)
```

Comment:
Add a quick unit test that a runtime without any auth handler modules will reject transactions.